src/mprispropertiesadaptor.cpp

```diff
@@ -171,8 +179,10 @@ QVariantMap MprisPropertiesAdaptor::GetAll(const QString &interface_name)
 
     QMap<QString, QString>::const_iterator it = getMap.constBegin();
     while (it != getMap.constEnd()) {
-        QVariant value = get(it.value());
-        result.insert(it.key(), value);
+        if (!m_maskedProperties.contains(it.key())) {
```

Comment:
I'm not sure if this is the best line to comment about this, but it seems introspecting will still show these properties. I don't think that's a huge problem but something to be aware of. It might be difficult to change.

Reply:
I didn't yet attempt to change this, but I had a look into it. It seems that changing introspection to skip these properties would require adding another adaptor that would have to work together with MprisPropertiesAdaptor too. In some ways I'd really like to hide the properties also from introspection as I think we can expect that some clients may use that to check if they are supported, but at the same time it's quite a bit of work and I don't really know if it's actually useful.